src/highdicom/seg/sop.py

```diff
@@ -99,9 +99,14 @@ def __init__(
         """
         Parameters
         ----------
-        source_images: Sequence[Dataset]
+        source_images: Sequence[pydicom.dataset.Dataset]
             One or more single- or multi-frame images (or metadata of images)
-            from which the segmentation was derived
+            from which the segmentation was derived. The images must have the
+            same dimensions (rows, columns) and orientation, have the same frame
+            of reference, and contain the same number of frames.
+            In case of multi-frame images that are tiled (e.g., VL Whole Slide
+            Microscopy Image instances), the images may be from more multiple
+            series as long as the other requirements are satisfied.
```

Comment:
I think you'll also need to update the `pixel_array` docstring, which describes the correspondence between the source images and the pixel array, since this is altered by your change.

Reply:
Addressed via a2c0d1d6342db23b20a3c3fab35056aceb8320f9